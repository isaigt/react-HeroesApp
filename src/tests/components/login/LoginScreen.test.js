import { mount } from 'enzyme';
import { AuthContext } from '../../../auth/authContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('Pruebas del <LoginScreen />', () => {
    const historyMock = {
        replace: jest.fn()
    }
    const contextMock = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }
    const wrapper = mount(
        <AuthContext.Provider value={ contextMock } >
            <LoginScreen history={ historyMock } />
        </AuthContext.Provider>
    );

    test('debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de realizar el dispatch y la navegacion', () => {
        wrapper.find('button').simulate('click');
        
        expect( contextMock.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'Isai'
            }
        });
        expect( historyMock.replace ).toHaveBeenCalledWith('/');
    });
});
