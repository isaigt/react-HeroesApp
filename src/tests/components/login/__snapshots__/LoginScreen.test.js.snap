// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del <LoginScreen /> debe de mostrarse correctamente 1`] = `
<div
  className="container mt-5 text-center"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Iniciar Sesion
  </button>
</div>
`;
